Content.Server/Flash/FlashSystem.cs

```diff
@@ -156,6 +156,10 @@ public void FlashArea(EntityUid source, EntityUid? user, float range, float dura
 
             foreach (var entity in _entityLookup.GetEntitiesInRange(transform.Coordinates, range))
             {
+                //cut to a circle.
+                if (MathF.Sqrt(MathF.Pow(mapPosition.X - Transform(entity).MapPosition.X, 2) + MathF.Pow(mapPosition.Y - Transform(entity).MapPosition.Y, 2)) > range)
+                    continue;
+
                 if (!flashableQuery.HasComponent(entity))
```

Comment:
Use the component queries and pass in the hashset.

Reply:
I honestly don't understand exactly what I'm supposed to do or how I'm supposed to do it.